### PR TITLE
Import io

### PR DESCRIPTION
**Pull request by connected-bduge**

### CODE REVIEW

File: custom.css [hunk 1]
```diff
@@ -38,5 +50,20 @@ div.title {
   overflow: hidden;
 }
 
-.case-step {
+#inputs {
```
Comment (connected-kkim) [inline custom.css]: better name difference between inputs, inputOuputs, etc

File: index.html [hunk 2]
```diff
@@ -8,6 +8,7 @@
       crossorigin="anonymous"
     />
     <link rel="stylesheet" href="custom.css" />
+    <script src="main-menu.js"></script>
   </head>
   <body>
     <h1 id="name">Alexa Testing Tool</h1>
```
Comment (connected-kkim) [inline index.html]: ChatterUI ? Chatter: Conversation Viewer? (Can you incorporate Chatter into the name so the use case is clear)

File: main-menu.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
+const { ipcRenderer, remote } = require("electron");
+const { dialog } = remote;
+var fs = require("fs");
+let json;
+let savedCases = {};
+ipcRenderer.on("reset", () => {
+  document.getElementById("name").innerHTML = "Boo";
```
Comment (connected-kkim) [inline main-menu.js]: Boo?

File: main-menu.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    if (fileNames === undefined) return;
+    let fileName = fileNames[0];
+    fs.readFile(fileName, "utf-8", (err, data) => {
+      if (err) throw err;
+      json = JSON.parse(data);
+      let test_cases = json.test_cases;
+      let inputs = json.inputs;
+      let outputs = json.outputs;
+
+      for (let key in test_cases) {
```
Comment (connected-kkim) [inline main-menu.js]: Repetitive code. create a function!

File: main-menu.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        } else {
+          description = outputs[key];
+        }
+        addInputOutput(key, description, "outputs");
+      }
+    });
+  });
+}
+
+function addSavedCase(name, cases) {
```
Comment (connected-kkim) [inline main-menu.js]: This function is pretty large, try to separate out larger functions like these. 

Also bonus points if you can start docstring for this code base as well. 👍

File: main-menu.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
+const { ipcRenderer, remote } = require("electron");
+const { dialog } = remote;
+var fs = require("fs");
```
Comment (connected-dpeng) [inline main-menu.js]: Is this ever reassigned? If not we can change it to a `const`.

File: main-menu.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  let inner_div = document.createElement("div");
+  inner_div.innerHTML = description;
+  div.appendChild(title);
+  div.appendChild(inner_div);
+  column.appendChild(div);
+
+  document.getElementById(type).appendChild(column);
+}
+
+function addOutput(name, description) {
```
Comment (connected-dpeng) [inline main-menu.js]: Duplicate function?

File: custom.css [hunk 8]
```diff
@@ -1,3 +1,9 @@
+h1 {
```
Comment (connected-dpeng) [inline custom.css]: I think you can group the `h1` and `h2` selectors like this:
```
h1, h2 {
  font-weight: bold;
}

File: main-menu.js [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+var fs = require("fs");
+let json;
+let savedCases = {};
+ipcRenderer.on("reset", () => {
+  document.getElementById("name").innerHTML = "Boo";
+});
+function browseFiles() {
+  dialog.showOpenDialog(fileNames => {
+    if (fileNames === undefined) return;
+    let fileName = fileNames[0];
```
Comment (connected-dpeng) [inline main-menu.js]: Is `fileName` ever reassigned? If not we can change it to a `const`.

File: main-menu.js [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+    fs.readFile(fileName, "utf-8", (err, data) => {
+      if (err) throw err;
+      json = JSON.parse(data);
+      let test_cases = json.test_cases;
+      let inputs = json.inputs;
+      let outputs = json.outputs;
```
Comment (connected-dpeng) [inline main-menu.js]: Can we change these variables to `const` as well?

File: main-menu.js [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+  div.appendChild(inner_div);
+
+  document.getElementById("savedCases").appendChild(div);
+  savedCases[name] = cases;
+}
+
+function addInputOutput(name, description, type) {
+  let column = document.createElement("div");
+  column.className = "col-12 inputOutputs";
+  let div = document.createElement("div");
```
Comment (connected-dpeng) [inline main-menu.js]: Having more descriptive names than `div` might make the function easier to follow.

File: custom.css [hunk 12]
```diff
@@ -38,5 +50,20 @@ div.title {
...
+
+.inputOutputs:hover {
+  background-color: deepskyblue;
+}
+
+#savedCases{
+  border: 1px solid black;
+}
+
+.it
```
Comment (connected-dpeng) [inline custom.css]: Selector with no rule?

**Pull request merged**